### PR TITLE
[PkgCI] Add presubmit regression testing on A100

### PR DESCRIPTION
**Pull request by Groverkss**

### CODE REVIEW

File: .github/workflows/pkgci_regression_test_a100.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  workflow_dispatch:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+
+jobs:
+  linux_x86_64:
+    name: Linux (x86_64)
+    runs-on: nodai-nvgpu-a100-x86-64
```
Comment (ScottTodd) [inline .github/workflows/pkgci_regression_test_a100.yml:23]: How many runners are there with this configuration? If the runner(s) have trouble keeping up with demand, we could change to T4 machines or postsubmit only testing. We switched which machines ran at which stages in the ci.yml jobs following inventory shortages, context in https://github.com/openxla/iree/pull/14765 and https://github.com/openxla/iree/issues/14169.

You could also mention in the PR description that these use new (persistent?) self-hosted runners, not the cloud auto-scaling runners used in other workflows.

Comment (ScottTodd) [reply .github/workflows/pkgci_regression_test_a100.yml:23]: Another thing that could help with usage would be connecting [`build_tools/github_actions/configure_ci.py`](https://github.com/openxla/iree/blob/main/build_tools/github_actions/configure_ci.py) to [`.github/workflows/pkgci.yml`](https://github.com/openxla/iree/blob/main/.github/workflows/pkgci.yml). That does some filtering so PRs that only change docs and other non-code files don't run all the builds/tests.

**Comment by Groverkss** [comment]
Not working on this anytime soon. Feel free to reopen if needed.

**Comment by ScottTodd** [comment]
I was planning on looking at this (testing CUDA and Vulkan on Nvidia GPUs) soon - possibly today. Thanks for starting this and the ping.

**Pull request closed without merging**